x/programs/rust/wasmlanche_sdk/src/types/mod.rs

```diff
@@ -1,73 +1,113 @@
-use crate::program::ProgramValue;
-use crate::store::ProgramContext;
+use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
 
+use crate::store::Context;
 /// A struct that enforces a fixed length of 32 bytes which represents an address.
-#[derive(Clone, Copy, PartialEq, Eq)]
-pub struct Address {
-    bytes: [u8; Self::LEN],
-}
+#[derive(Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Debug)]
+pub struct Address(Bytes32);
 
 impl Address {
     pub const LEN: usize = 32;
     // Constructor function for Address
     pub fn new(bytes: [u8; Self::LEN]) -> Self {
-        Self { bytes }
+        Self(Bytes32::new(bytes))
     }
     pub fn as_bytes(&self) -> &[u8] {
-        &self.bytes
+        self.0.as_bytes()
     }
 }
 
-impl From<String> for ProgramValue {
-    fn from(value: String) -> Self {
-        ProgramValue::StringObject(value)
+impl From<i64> for Address {
+    fn from(value: i64) -> Self {
+        Self(Bytes32::from(value))
     }
 }
 
-impl From<&str> for ProgramValue {
-    fn from(value: &str) -> Self {
-        ProgramValue::StringObject(String::from(value))
+/// A struct representing a fixed length of 32 bytes.
+/// This can be used for passing strings to the host. It caps the string at 32 bytes,
+#[derive(Clone, Copy, PartialEq, Eq, Serialize, Deserialize, Debug)]
+pub struct Bytes32([u8; Self::LEN]);
```

Comment:
Not for this PR, but `Bytes32` and `Address` are the exact same thing; why have separate types?

Reply:
Was thinking we would use bytes32 for Strings that we want to pass across go/rust